gateway/src/api/latest.rs

```diff
@@ -255,23 +259,53 @@ async fn route_project(
         (status = 500, description = "Server internal error.")
     )
 )]
-async fn get_status(State(RouterState { sender, .. }): State<RouterState>) -> Response<Body> {
-    let (status, body) = if sender.is_closed() || sender.capacity() == 0 {
-        (
-            StatusCode::INTERNAL_SERVER_ERROR,
-            StatusResponse::unhealthy(),
-        )
+async fn get_status(
+    State(RouterState {
+        sender, service, ..
+    }): State<RouterState>,
+) -> Response<Body> {
+    let mut statuses = Vec::new();
+    // Compute gateway status.
+    if sender.is_closed() || sender.capacity() == 0 {
+        statuses.push((SHUTTLE_GATEWAY_VARIANT, StatusResponse::unhealthy()));
     } else if sender.capacity() < WORKER_QUEUE_SIZE - SVC_DEGRADED_THRESHOLD {
-        (StatusCode::OK, StatusResponse::degraded())
+        statuses.push((SHUTTLE_GATEWAY_VARIANT, StatusResponse::degraded()));
+    } else {
+        statuses.push((SHUTTLE_GATEWAY_VARIANT, StatusResponse::healthy()));
+    };
+
+    // Compute provisioner status.
+    let provisioner_status = if let Ok(channel) = service.provisioner_host().connect().await {
+        let channel = ServiceBuilder::new().service(channel);
+        let mut provisioner_client = ProvisionerClient::new(channel);
+        if provisioner_client.health_check(Ping {}).await.is_ok() {
+            StatusResponse::healthy()
+        } else {
+            StatusResponse::unhealthy()
+        }
     } else {
-        (StatusCode::OK, StatusResponse::healthy())
+        StatusResponse::unhealthy()
     };
 
-    let body = serde_json::to_vec(&body).unwrap();
+    statuses.push(("shuttle-provisioner", provisioner_status));
+
+    // Compute auth status.
+    let auth_status = if let Ok(response) = reqwest::get(service.auth_uri().to_string()).await {
```

Comment:
Nit: should we rather have a reqwest client in the router state, or in a lazy_static? So we don't create it for each request.

Reply:
tip: OnceCell / OnceLock in Rust 1.70 😉

Reply:
Thanks guys! Used a `OnceLock` because it might be shared between threads.